src/librustc/hir/lowering/expr.rs

```diff
@@ -394,30 +394,49 @@ impl LoweringContext<'_> {
 
     fn lower_expr_try_block(&mut self, body: &Block) -> hir::ExprKind {
         self.with_catch_scope(body.id, |this| {
-            let unstable_span = this.mark_span_with_reason(
+            let mut block = this.lower_block(body, true).into_inner();
+
+            let tail_expr = block.expr.take().map_or_else(
+                || {
+                    let unit_span = this.mark_span_with_reason(
+                        DesugaringKind::TryBlock,
+                        this.sess.source_map().end_point(body.span),
```

Comment:
Would be good to leave a comment here (and elsewhere in the method).

Reply:
Ok.